### PR TITLE
[Fix] Allow creating stickers with no description

### PR DESCRIPTION
**Pull request by Misha-133**
This PR allows creating custom stickers without a description

###Changes
- **BREAKING** make `description` parameter optional in `CreateStickerAsync` methods

### CODE REVIEW

File: src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs [hunk 1]
```diff
@@ -705,49 +705,67 @@ public static async Task<GuildEmote> GetEmoteAsync(IGuild guild, BaseDiscordClie
...
+        public static async Task<API.Sticker> CreateStickerAsync(BaseDiscordClient client, IGuild guild, string name, Image image, IEnumerable<string> tags,
+            string description = null, RequestOptions options = null)
         {
             Preconditions.NotNull(name, nameof(name));
-            Preconditions.NotNull(description, nameof(description));
+
+            if (description is not null)
+            {
+                Preconditions.AtLeast(description.Length, 2, nameof(description));
+                Preconditions.AtMost(description.Length, 100, nameof(name));
```
Comment (dimson-n) [inline src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs]:
```suggestion
                Preconditions.AtMost(description.Length, 100, nameof(description));
```

Comment (Misha-133) [reply src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs]: ah lol, nice catch
👍

File: src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs [hunk 2]
```diff
@@ -705,49 +705,67 @@ public static async Task<GuildEmote> GetEmoteAsync(IGuild guild, BaseDiscordClie
...
+            }
+
+            var tagString = string.Join(", ", tags);
+
+            Preconditions.AtLeast(tagString.Length, 1, nameof(description));
```
Comment (dimson-n) [inline src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs]:
```suggestion
            Preconditions.AtLeast(tagString.Length, 1, nameof(tagString));
```

File: src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs [hunk 3]
```diff
@@ -705,49 +705,67 @@ public static async Task<GuildEmote> GetEmoteAsync(IGuild guild, BaseDiscordClie
...
+            Preconditions.AtMost(tagString.Length, 200, nameof(name));
```
Comment (dimson-n) [inline src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs]:
```suggestion
            Preconditions.AtMost(tagString.Length, 200, nameof(tagString));
```

File: src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs [hunk 4]
```diff
@@ -705,49 +705,67 @@ public static async Task<GuildEmote> GetEmoteAsync(IGuild guild, BaseDiscordClie
...
-            Preconditions.AtLeast(description.Length, 2, nameof(description));
 
             Preconditions.AtMost(name.Length, 30, nameof(name));
-            Preconditions.AtMost(description.Length, 100, nameof(name));
+
+
+            if (description is not null)
+            {
+                Preconditions.AtLeast(description.Length, 2, nameof(description));
+                Preconditions.AtMost(description.Length, 100, nameof(name));
```
Comment (dimson-n) [inline src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs]:
```suggestion
                Preconditions.AtMost(description.Length, 100, nameof(description));
```

File: src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs [hunk 5]
```diff
@@ -705,49 +705,67 @@ public static async Task<GuildEmote> GetEmoteAsync(IGuild guild, BaseDiscordClie
...
+            }
+            
+            var tagString = string.Join(", ", tags);
+
+            Preconditions.AtLeast(tagString.Length, 1, nameof(description));
```
Comment (dimson-n) [inline src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs]:
```suggestion
            Preconditions.AtLeast(tagString.Length, 1, nameof(tagString));
```

File: src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs [hunk 6]
```diff
@@ -705,49 +705,67 @@ public static async Task<GuildEmote> GetEmoteAsync(IGuild guild, BaseDiscordClie
...
+            Preconditions.AtMost(tagString.Length, 200, nameof(name));
```
Comment (dimson-n) [inline src/Discord.Net.Rest/Entities/Guilds/GuildHelper.cs]:
```suggestion
            Preconditions.AtMost(tagString.Length, 200, nameof(tagString));
```

**Review by csmir — APPROVED** [review]

**Pull request merged**